### PR TITLE
disttask/ddl: fix pause succeed task then run a new task

### LINKED ISSUE #47805: ddl: TestAddIndexDistPauseAndResume failed
## Bug Report

Please answer these questions before submitting your issue. Thanks!
TestAddIndexDistPauseAndResume takes a long time.
### 2. What did you expect to see? (Required)
Test passed stable.
The last dist task run once.
### 3. What did you see instead (Required)
img_v2_6a9348b9-defa-4d6a-8ae4-eb8fed2f471g
The last dist task run twice.
### 4. What is your TiDB version? (Required)
master

### PR DESCRIPTION
**Pull request by ywqzzy**
### What problem does this PR solve?

Issue Number: close #47805 

Problem Summary:
The pause ddl job was triggered after the task was successfully completed, and at this point, the task had already been moved to the history table. After resuming the ddl, the table was not queried from the history table, resulting in the framework reopening the task and running it all over again.
### What is changed and how it works?
1. Use Channel to make sure the pause cmd actually works.
2. Fetch task from global_task table and history table to make sure the task only run once.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  >   

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
optimize the pause&resume process when a dist task succeed.
```

### CODE REVIEW

File: pkg/ddl/index.go [hunk 1]
```diff
@@ -2066,19 +2069,27 @@ func (w *worker) executeDistGlobalTask(reorgInfo *reorgInfo) error {
...
+	// Need to fetch global task by taskKey in tidb_global_task and tidb_global_task_history tables.
+	// When pausing the related ddl job, it is possible that the global task with taskKey is succeed and in tidb_global_task_history.
+	// As a result, when resuming the related ddl job,
+	// it is necessary to check task exits in tidb_global_task and tidb_global_task_history tables.
 	taskManager, err := storage.GetTaskManager()
 	if err != nil {
 		return err
 	}
-	task, err := taskManager.GetGlobalTaskByKey(taskKey)
+	task, err := taskManager.GetGlobalTaskByKeyWithHistory(taskKey)
```
Comment (D3Hunter) [inline pkg/ddl/index.go:2081]: why paused task moved into history?

Comment (ywqzzy) [reply pkg/ddl/index.go:2081]: > why paused task moved into history?

It has possiblity that the dist task was finished but the ddl job was paused.

**Review by D3Hunter — APPROVED** [review]

**Review by Benjamin2037 — APPROVED** [review]
LGTM

**Pull request merged**